### PR TITLE
MAGN-10617 Can't update a package

### PR DESCRIPTION
**Pull request by NightHawk-Michael**
### Purpose

Cross-merge from https://github.com/DynamoDS/Dynamo/pull/7114

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Follows [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) standards (*No change of public methods or types etc..*)

### Reviewers
@Benglin

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs [hunk 1]
```diff
@@ -63,13 +63,14 @@ public TermsOfUseHelper(TermsOfUseHelperParams touParams)
             authenticationManager = touParams.AuthenticationManager;
         }
 
-        public void Execute()
+
+        public void Execute(bool preventReentrant = true)
```
Comment (ke-yu) [inline src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs]: Break binary compatibility, though I don't think anyone else will use this class.

**Comment by Benglin** [comment]
Please see [the other pull request](https://github.com/DynamoDS/Dynamo/pull/7144) for additional review comments.

**Comment by Benglin** [comment]
Thanks @NightHawk-Michael for the fix, please let me know once it is tested on `master` branch, then we'll merge this into `RC` branch.

**Comment by Benglin** [comment]
As discussed with @monikaprabhu based on preliminary tests, I'm merging this into RC branch for better test coverage.

**Pull request merged**